Create a shared progress stepper #1

ravighevariya10 opened the pull request:
Created a shared Progress Stepper component

ravighevariya10 committed:
Created a shared Progressstepper

ravighevariya10 committed:
updated the onClick method

ravighevariya10 committed:
printing the steps

ravighevariya10 committed:
updated the step-name displaying logic

ravighevariya10 committed:
updated the onClick method

ravighevariya10 committed:
updated the onClick method

ravighevariya10 committed:
added new prop tabs

ravighevariya10 committed:
added new prop 'getStepIcon'

ravighevariya10 committed:
updated the onClick method

ravighevariya10 committed:
updated the onClick method

ravighevariya10 committed:
updated the getStepIcon method

ravighevariya10 committed:
updated the story

ravighevariya10 committed:
updated the stepButton style

ravighevariya10 committed:
updated the setting activeStep logic

ravighevariya10 committed:
updated the activeStep logic

ravighevariya10 committed:
printing the console log

ravighevariya10 committed:
updated the activeStep logic

ravighevariya10 committed:
1. Deleted the unnecessary folder
2. Updated the logic of the stepper styling

ravighevariya10 committed:
1. Added the propTypes
2. Updated the logic for steps accessibility

YazidZidane commented on package.json:
```diff
@@ -37,10 +37,13 @@
     "chroma-js": "^3.0.0",
     "mapbox-gl": "^3.6.0",
+    "react-redux": "^9.1.2",
+    "react-router-dom": "^7.0.2",
```
Can you remove these changes and changes in package-lock if they will not be used?

YazidZidane commented on src/Progressstepper/progressstepper.stories.jsx:
```diff
@@ -0,0 +1,47 @@
+  component: PSAProgressstepper,
+  tags: ["autodocs"],
+  parameters: {
+    layout: "centered",
```
Can you add `title: "Functional/Stepper"` before the `component:`line, and update layout to `fullscreen`?

YazidZidane commented on src/Progressstepper/progressstepper.jsx:
```diff
@@ -0,0 +1,228 @@
+  /**
+   * Array of tabs or additional data that is associated with each step.
+   */
+  tabs: PropTypes.arrayOf(PropTypes.any).isRequired,
```
Can you give example about this prop?

ravighevariya10 replied:
This prop specifies the tab the app navigates to when the stepButton is clicked.

YazidZidane replied:
So it's also an array associated with `steps[]` right? I think this might not be necessary to be passed in the component

ravighevariya10 replied:
Yup you r right

YazidZidane reviewed (commented):
Can you also update the component name to `Stepper`?

YazidZidane commented on src/Progressstepper/progressstepper.stories.jsx:
```diff
@@ -0,0 +1,47 @@
+      activeStep: 0
+    },
+    boxProps: {
+      sx: { backgroundColor: '#F5F5F5', opacity: 0.9, width: '100%' }
```
Can you remove the bgcolor and opacity since they are already defined in the component?

ravighevariya10 committed:
1. Removed the unnecessary dependencies
2. Renamed the component name to 'Stepper'
3. Gave a title to the story folder
4. Removed the boxProps from the stories as it has the same default values

ravighevariya10 committed:
updated the ActiveButton style

YazidZidane commented on package.json:
```diff
@@ -37,6 +37,7 @@
     "@fontsource/ibm-plex-sans": "^5.1.0",
     "@mapbox/mapbox-gl-draw": "^1.4.3",
     "@mapbox/mapbox-gl-geocoder": "^5.0.2",
+    "@reduxjs/toolkit": "^2.4.0",
```
Can you also remove this?

ravighevariya10 committed:
Updated the active-step styling

ravighevariya10 committed:
1. Removed the unnecessary dependency"
2. Removed an unnecessary prop named 'tabs'

ravighevariya10 committed:
added the data-test to be set in the StepButton instead of having it as prop

YazidZidane committed:
add mobile stepper

remove outer box, update prop names

YazidZidane committed:
rename file

YazidZidane committed:
Merge branch 'develop' into progress-stepper
